library/cc/bridge_utility.cc

```diff
@@ -0,0 +1,111 @@
+#include "bridge_utility.h"
+
+// NOLINT(namespace-envoy)
+
+// TODO(crockeo): we always copy memory across boundaries; consider allowing for moves and/or
+// shared ownership w/ reference counting via envoy-mobile's release callbacks
+
+#include <sstream>
+
+envoy_data buffer_as_envoy_data(const std::vector<uint8_t>& data) {
```

Comment:
this whole file is going to be a little messy since i hacked it together quickly

i imagine there's already code that translates between the C types and managed C++ types. i'd prefer to reuse that rather than reconstructing complexity here

Reply:
there's also a bunch of overhead from copying data across the platform/bridge layers. i'm going to come back and fix it later, but i thought i'd get it correct first, and make it right later.